trunk/SUAVE/Components/Energy/Converters/Compression_Nozzle.py

```diff
@@ -104,32 +106,61 @@ def compute(self,conditions):
         gamma   = conditions.freestream.isentropic_expansion_factor
         Cp      = conditions.freestream.specific_heat_at_constant_pressure
         Po      = conditions.freestream.pressure
+        Mo      = conditions.freestream.mach_number
         R       = conditions.freestream.universal_gas_constant
         
         #unpack from inpust
         Tt_in   = self.inputs.stagnation_temperature
         Pt_in   = self.inputs.stagnation_pressure
         
         #unpack from self
-        pid     =  self.pressure_ratio
-        etapold =  self.polytropic_efficiency
+        pid                     =  self.pressure_ratio
+        etapold                 =  self.polytropic_efficiency
+        compressibility_effects =  self.compressibility_effects
+
         
         #Method to compute the output variables
         
         #--Getting the output stagnation quantities
-        Pt_out  = Pt_in*pid
         Tt_out  = Tt_in*pid**((gamma-1)/(gamma*etapold))
         ht_out  = Cp*Tt_out
         
-        # in case pressures go too low
-        if np.any(Pt_out<Po):
-            warn('Pt_out goes too low',RuntimeWarning)
-            Pt_out[Pt_out<Po] = Po[Pt_out<Po]
+
+        if compressibility_effects :
+            #compute the output Mach number, static quantities and the output velocity
+            i_low  = Mo <= 1.0
+            i_high = Mo > 1.0
+            
+            #initializing the arrays
+            Mach     = 1.0 * Pt_in/Pt_in
+            T_out    = 1.0 * Pt_in/Pt_in
+            Mo       = Mo * Pt_in/Pt_in
+            Pt_out   = 1.0 * Pt_in/Pt_in
+            P_out    = 1.0 * Pt_in/Pt_in
+            
+            #-- Inlet Mach <= 1.0, isentropic relations
+            Pt_out[i_low]  = Pt_in[i_low]*pid
+            Mach[i_low]    = np.sqrt( (((Pt_out[i_low]/Po[i_low])**((gamma-1.)/gamma))-1.) *2./(gamma-1.) )
+            T_out[i_low]   = Tt_out[i_low]/(1+(gamma-1)/2*Mach[i_low]*Mach[i_low])
+            
+            #-- Inlet Mach > 1.0, normal shock
+            Mach[i_high]   = np.sqrt((1.+(gamma-1.)/2.*Mo[i_high]**2.)/(gamma*Mo[i_high]**2-(gamma-1.)/2.))
+            T_out[i_high]  = Tt_out[i_high]/(1.+(gamma-1.)/2*Mach[i_high]*Mach[i_high])
+            Pt_out[i_high] = Pt_in[i_high]*((((gamma+1.)*(Mo[i_high]**2.))/((gamma-1.)*Mo[i_high]**2.+2.))**(gamma/(gamma-1.)))*((gamma+1.)/(2.*gamma*Mo[i_high]**2.-(gamma-1.)))**(1./(gamma-1.))
+            P_out[i_high]  = Pt_out[i_high]*(1.+(2.*gamma/(gamma+1.))*(Mach[i_high]**2.-1.))
```

Comment:
By nozzle geometry I suppose you're referring to pid? Because in that case, I can apply the nozzle pressure loss and take it into account when computing the pressure recovery factor

Reply:
I suppose so regarding pid. Have you done this somewhere or are you just saying it's possible?

Reply:
For the shock analysis, I considered the pid effect negligible compared with the pressure recovery factor, which is far superior. However, I have now included it in the latest commit. The effect should be minimum

And to answer your question, I have not done it anywhere else

Reply:
Okay looks good. I'd say in general these kind of assumptions are fine as long as they're noted in the docstring or comments.

Reply:
Got it, will do so from now on